pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -185,7 +185,7 @@ public class ServerCnx extends PulsarHandler implements TransportCnx {
     private String clientVersion = null;
     private int nonPersistentPendingMessages = 0;
     private final int maxNonPersistentPendingMessages;
-    private String originalPrincipal = null;
+    String originalPrincipal = null;
```

Comment:
It's better to add a @VisibleForTesting annotation here. And we should only expose the getter, not the setter.

```java
@VisibleForTesting
String getOriginalPrincipal() {
    return originalPrincipal;
}
```